banks-interface/src/lib.rs

```diff
@@ -39,12 +39,25 @@ pub struct TransactionSimulationDetails {
     pub return_data: Option<TransactionReturnData>,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionMetadata {
+    pub log_messages: Vec<String>,
+    pub compute_units_consumed: u64,
+}
```

Comment:
Could you add return data and inner instructions to this? I've definitely had times where I needed return data, and no cases where I needed inner instructions, but it doesn't seem ridiculous to want that too.

Reply:
Will add return data but I'd like to wait on adding inner instruction a bit longer until we can pipe through the number of compute units and return data for each inner instruction because those additions might require changing the data structure in a breaking way.

Reply:
Ah ok, makes sense!

Reply:
Sorry, did you mean you'll add return data in this PR or a future one?

Reply:
I meant this one, but I must have lost that change somehow. Just added, thanks!